server/app/service/ScalaTestRunner.scala

```diff
@@ -18,6 +24,22 @@ object ScalaTestRunner {
   val traitDefPattern = """trait\s*([\w\$]*)""".r
   val defaultImports = "import org.scalatest._"
 
+  def apply(suiteClass: Class[Suite], solutionTrait: Class[AnyRef])(solution: String)(implicit s: Scheduler): Observable[Line] = {
```

Comment:
man, I really do not like this modification in this class. Let me explain why:
1) This service responsibility is to ONLY run test via reflection. That is it.
2) There were overloaded execSuite methods by intention. They are different: one to run runtime tests and other to run code from text.
3) Now we have 3 rd method which is apply and this is same as:
def execSuite(solution: String, suiteClass: Class[Suite], solutionTrait: Class[AnyRef]): String =

What I propose is to not add here additional responsibilities. And monifu reactive library is one more responsibility here! Can you just add proxy service to call execSuite from it and decorate ScalaTestRunner with Observable?